File: Data Structures & Algorithms/first-missing-positive/submission-2.cpp
class Solution {
public:
    int firstMissingPositive(vector<int>& nums) {
        for (int i = 0; i < nums.size(); i++) {
            while (1 <= nums[i] && nums[i] <= nums.size() && nums[i] != i + 1 && nums[nums[i] - 1] != nums[i]) {
                swap(nums[i], nums[nums[i] - 1]);
            }
        }

        for (int j = 0; j < nums.size(); j++) {
            if (nums[j] != j + 1) return j + 1;
        }
        return nums.size() + 1;
    }
};

File: Data Structures & Algorithms/maximum-subarray/submission-2.py
class Solution:
    def maxSubArray(self, nums: List[int]) -> int:
        l = 0
        currSum = 0
        maxSum = nums[0]

        for r in range(len(nums)):
            currSum += nums[r]
            maxSum = max(currSum, maxSum)
            while currSum < 0:
                currSum -= nums[l]
                l += 1

        return maxSum
                